### PR TITLE
Rename rest of the dynamodb flags

### PR DESCRIPTION
**Pull request by gouthamve**
Fixes https://github.com/cortexproject/cortex/issues/2254

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: integration/configs.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ storage:
       dynamodb_url: {{.DynamoDBURL}}
 
 table_manager:
-  dynamodb_poll_interval: 1m
+  poll_interval: 1m
```
Comment (tomwilkie) [inline integration/configs.go]: nit: whitespace to line up values?

File: pkg/chunk/table_manager.go [hunk 2]
```diff
@@ -47,8 +47,8 @@ func newTableManagerMetrics(r prometheus.Registerer) *tableManagerMetrics {
 
 	m.tableCapacity = prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: "cortex",
-		Name:      "dynamo_table_capacity_units",
-		Help:      "Per-table DynamoDB capacity, measured in DynamoDB capacity units.",
+		Name:      "table_capacity_units",
```
Comment (tomwilkie) [inline pkg/chunk/table_manager.go]: We discussed this in #2307, it is actually dynamodb specific.

File: pkg/chunk/table_manager.go [hunk 3]
```diff
@@ -161,21 +161,21 @@ func (cfg *TableManagerConfig) RegisterFlags(f *flag.FlagSet) {
...
+	f.Int64Var(&cfg.ProvisionedWriteThroughput, argPrefix+".write-throughput", 1000, "Table default write throughput. Supported by DynamoDB")
+	f.Int64Var(&cfg.ProvisionedReadThroughput, argPrefix+".read-throughput", 300, "Table default read throughput. Supported by DynamoDB")
+	f.BoolVar(&cfg.ProvisionedThroughputOnDemandMode, argPrefix+".enable-ondemand-throughput-mode", false, "Enables on demand throughput provisioning for the storage provider (if supported). Applies only to tables which are not autoscaled. Supported by DynamoDB")
+	f.Int64Var(&cfg.InactiveWriteThroughput, argPrefix+".inactive-write-throughput", 1, "Table write throughput for inactive tables. Supported by DynamoDB")
+	f.Int64Var(&cfg.InactiveReadThroughput, argPrefix+".inactive-read-throughput", 300, "Table read throughput for inactive tables. Supported by DynamoDB")
+	f.BoolVar(&cfg.InactiveThroughputOnDemandMode, argPrefix+".inactive-enable-ondemand-throughput-mode", false, "Enables on demand throughput provisioning for the storage provider (if supported). Applies only to tables which are not autoscaled. Supported by DynamoDB")
```
Comment (tomwilkie) [inline pkg/chunk/table_manager.go]: These changes look a little redundant, why are they here?

File: pkg/chunk/table_manager.go (hunk 2 above)
Comment (gouthamve) [reply pkg/chunk/table_manager.go]: https://cloud-native.slack.com/archives/CCYDASBLP/p1585577302112200

> @bboreham: Today it is only on DynamoDB, but it could be implemented for another DB, e.g. CosmosDB at Azure.

Comment (tomwilkie) [reply pkg/chunk/table_manager.go]: Roger.  I don't mind either way.

File: CHANGELOG.md [hunk 4]
```diff
@@ -45,6 +45,11 @@
   - `-redis.password` --> `-store.chunks-cache.redis.password`
   - `-redis.timeout` --> `-store.chunks-cache.redis.timeout`
 * [CHANGE] Rename the `-store.chunk-cache-stubs` to `-store.chunks-cache.cache-stubs` to be more inline with above.
+* [CHANGE] Change prefix of flags `-dynamodb.periodic-table.*` to `-table-manager.index-table.*`. #2359
+* [CHANGE] Change prefix of flags `-dynamodb.chunk-table.*` to `-table-manager.chunk-table.*`. #2359
+* [CHANGE] Change the following flags: #2359
+  - `-dynamodb.poll-interval` --> `-table-manager.poll-interval`
+  - `-dynamodb.periodic-table.grace-period` --> `table-manager.periodic-table.grace-period`
```
Comment (pracucci) [inline CHANGELOG.md]:
```suggestion
  - `-dynamodb.periodic-table.grace-period` --> `-table-manager.periodic-table.grace-period`
```

File: docs/configuration/config-file-reference.md [hunk 5]
```diff
@@ -1028,315 +1028,319 @@ The `table_manager_config` configures the Cortex table-manager.
 # CLI flag: -table-manager.retention-period
 [retention_period: <duration> | default = 0s]
 
-# How frequently to poll DynamoDB to learn our capacity.
-# CLI flag: -dynamodb.poll-interval
-[dynamodb_poll_interval: <duration> | default = 2m0s]
+# How frequently to poll backend to learn our capacity.
+# CLI flag: -table-manager.poll-interval
+[poll_interval: <duration> | default = 2m0s]
```
Comment (pracucci) [inline docs/configuration/config-file-reference.md]: Remember to mention this YAML config rename (`dynamodb_poll_interval` to `poll_interval`) in the changelog

File: docs/configuration/config-file-reference.md [hunk 6]
```diff
@@ -1028,315 +1028,319 @@ The `table_manager_config` configures the Cortex table-manager.
...
 [creation_grace_period: <duration> | default = 10m0s]
 
 index_tables_provisioning:
   # Enables on demand throughput provisioning for the storage provider (if
-  # supported). Applies only to tables which are not autoscaled
-  # CLI flag: -dynamodb.periodic-table.enable-ondemand-throughput-mode
+  # supported). Applies only to tables which are not autoscaled. Supported by
+  # DynamoDB
+  # CLI flag: -table-manager.index-table.enable-ondemand-throughput-mode
   [provisioned_throughput_on_demand_mode: <boolean> | default = false]
```
Comment (pracucci) [inline docs/configuration/config-file-reference.md]: I personally find the YAML config name quite confusing, compared to the CLI flag. Have you considered rename `provisioned_throughput_on_demand_mode` to `enable_ondemand_throughput_mode`?

_Same comment applies to `inactive_throughput_on_demand_mode` and same settings to `chunk_tables_provisioning` (to keep it consistent)_

File: integration/backward_compatibility_test.go [hunk 7]
```diff
@@ -73,7 +75,7 @@ func runBackwardCompatibilityTestWithChunksStorage(t *testing.T, previousImage s
 
 	// Start other Cortex components (ingester running on previous version).
 	ingester1 := e2ecortex.NewIngester("ingester-1", consul.NetworkHTTPEndpoint(), flagsForOldImage, previousImage)
-	distributor := e2ecortex.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), flagsForOldImage, "")
+	distributor := e2ecortex.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), ChunksStorageFlags, "")
```
Comment (pracucci) [inline integration/backward_compatibility_test.go]: Why this change? This change the meaning of the test. We're not testing anymore an old distributor anymore.

File: pkg/chunk/table_manager.go (hunk 2 above)
Comment (pracucci) [reply pkg/chunk/table_manager.go]: If we keep this change, it should be mentioned in the changelog, otherwise will be difficult to notice for the final user.

File: integration/backward_compatibility_test.go (hunk 7 above)
Comment (pracucci) [reply integration/backward_compatibility_test.go]: Sorry. I misread it. We're actually using the new image, so the previous code looks incorrect. Thanks for fixing!

File: CHANGELOG.md [hunk 8]
```diff
@@ -45,6 +45,11 @@
   - `-redis.password` --> `-store.chunks-cache.redis.password`
   - `-redis.timeout` --> `-store.chunks-cache.redis.timeout`
 * [CHANGE] Rename the `-store.chunk-cache-stubs` to `-store.chunks-cache.cache-stubs` to be more inline with above.
+* [CHANGE] Change prefix of flags `-dynamodb.periodic-table.*` to `-table-manager.index-table.*`. #2359
```
Comment (bboreham) [inline CHANGELOG.md]: Weren't these already removed by #2221 ?

Comment (bboreham) [reply CHANGELOG.md]: Oh I get it - these are the scaling flags.  So the description here could be made clearer by saying that.
